### PR TITLE
pypyr Slack Plug-In: First Blood

### PR DESCRIPTION
**Pull request by 345thomas**

### CODE REVIEW

File: pypyrslack/steps/send.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        AssertionError: When any of slackToken, slackChannel or slackText do
+                        not have values.
+        SlackSendError: Something went wrong sending to slack. Be sure your
+                        api token has access to the channel.
+    """
+    logger.debug("started")
+    context.assert_keys_have_values(__name__,
+                                    'slackToken',
+                                    'slackChannel',
+                                    'slackText')
```
Comment (345andrew) [inline pypyrslack/steps/send.py]: Is the Slack text always pre-formatted, or does it attempt to format from context?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+interpolation. This substitutes anything between curly braces with the context
+value for that key. For example, if your context looked like this:
+
+.. code-block:: yaml
+
+  arbitraryValue: pypyrchannel
+  arbitraryText: down the
+  moreArbText: wild
+  slackChannel: "#{arbitraryValue}"
+  slackText: "piping {arbitraryText} the valleys {moreArbText}"
```
Comment (345andrew) [inline README.rst]: I think this answers my question on the step.
pypyng down the valleys?

**Comment by 345andrew** [comment]
I also like the Stallone reference in the PR title

**Comment by 345thomas** [comment]
It was actually an @andrew reference - I was thinking of those famous "first cut"s of yours :-D ;-)

File: README.rst (hunk 2 above)
Comment (345thomas) [inline README.rst]: William Blake. I'll keep the spelling to piping in his honour.

Yes, the output text would be "piping down the valleys wild"

Comment (345thomas) [inline README.rst]: Which I'm sure you recognized from the well loved poem from his visionary Songs of Innocence & Experience.

**Pull request merged**